compiler/rustc_data_structures/src/sync.rs

```diff
@@ -381,55 +385,106 @@ impl<K: Eq + Hash, V: Eq, S: BuildHasher> HashMapExt<K, V> for HashMap<K, V, S>
     }
 }
 
-#[derive(Debug)]
-pub struct Lock<T>(InnerLock<T>);
+pub struct Lock<T> {
+    single_thread: bool,
```

Comment:
Shouldn't loading `PARALLEL` with relaxed ordering (or maybe a non-atomic read) be just as efficient? That avoids having to copy around the `bool` in every lock.